src/main/java/org/openstreetmap/atlas/checks/base/CheckResourceLoader.java

```diff
@@ -83,29 +96,59 @@ public CheckResourceLoader(final Configuration configuration)
         this.configuration = configuration;
     }
 
+    public Configuration getConfiguration()
+    {
+        return this.configuration;
+    }
+
     /**
-     * Loads checks that are enabled by configuration
+     * Get configuration for a specific country, overriding for country specific overrides and group
+     * specific overrides
      *
-     * @param <T>
-     *            check type
-     * @return a {@link Set} of checks
+     * @param country
+     *            country string
+     * @return {@link Configuration}
      */
-    public <T extends Check> Set<T> loadChecks()
+    public Configuration getConfigurationForCountry(final String country)
+    {
+        Configuration specializedConfiguration = this.configuration
+                .configurationForKeyword(country);
+
+        if (this.countryGroups.containsKey(country))
+        {
+            for (final String group : this.countryGroups.get(country))
+            {
+                specializedConfiguration = specializedConfiguration.configurationForKeyword(group);
+            }
+        }
+        return specializedConfiguration;
+    }
+
+    public <T extends Check> Set<T> loadChecks(final Predicate<Class> isEnabled)
+    {
+        return loadChecks(isEnabled, this.configuration);
+    }
+
+    public <T extends Check> Set<T> loadChecks(final Configuration configuration)
     {
-        return loadChecks(this::isEnabledByConfiguration);
+        return loadChecks(this::isEnabledByConfiguration, configuration);
     }
 
     /**
      * Loads checks that are enabled by some other means, defined by {@code isEnabled}
      *
      * @param isEnabled
      *            {@link Predicate} used to determine if a check is enabled
+     * @param specificConfiguration
+     *            {@link Configuration} used to loadChecks if not the one used to initialize
+     *            {@link CheckResourceLoader}
      * @param <T>
      *            check type
      * @return a {@link Set} of checks
      */
     @SuppressWarnings("unchecked")
-    public <T extends Check> Set<T> loadChecks(final Predicate<Class> isEnabled)
+    public <T extends Check> Set<T> loadChecks(final Predicate<Class> isEnabled,
+            final Configuration specificConfiguration)
```

Comment:
`specificConfiguration` name is misleading. In what way it is specific?

Reply:
It is the configuration made specifically for a country.

Reply:
This method is loading checks for given configuration. It is oblivious to overrides. So I'd suggest leaving naming as-is.